place-work/v1alpha1/types.go

```diff
@@ -0,0 +1,134 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	work "open-cluster-management.io/api/work/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// PlaceManifestWorkSpec defines the desired state of PlaceManifestWork
+type PlaceManifestWorkSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ManifestWorkSpec is the ManifestWorkSpec that will be used to generate a per-cluster ManifestWork
+	ManifestWorkSpec work.ManifestWorkSpec `json:"manifestWorkSpec"`
+
+	// PacementRef is the name of the Placement resource, from which a PlacementDecision will be found and used
+	// to distribute the ManifestWork
+	PlacementRef *corev1.LocalObjectReference `json:"placementRef,omitempty"`
+}
+
+// PlaceManifestWorkStatus defines the observed state of PlaceManifestWork
+type PlaceManifestWorkStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Conditions contains the different condition statuses for distrbution of ManifestWork resources
+	// Valid condition types are:
+	// 1. AppliedManifestWorks represents ManifestWorks have been distributed as per placement All, Partial, None, Problem
+	// 2. PlacementRefValid
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ManifestWorkDelivery tracks each ManifestWork that was created,
+	// including that it was applied and its overall status
+	PlacedManifestWork []PlacedManifestWork `json:"placedManifestWork,omitempty"`
+
+	// Summary that reflects all relevant ManifestWorks
+	PlacedManifestWorkSummary []PlacedManifestWorkSummary `json:"summary"`
+}
+
+type PlacedManifestWorkSummary struct {
+	Total       int `json:"total"`
+	Progressing int `json:"progressing"`
+	Available   int `json:"available"`
+	Degraded    int `json:"degraded"`
+	Applied     int `json:"Applied"`
+}
+
+type ManifestWorkStatus string
+
+type PlacedManifestWork struct {
+
+	// Work is an objectReference to the actual ManifestWork resource
+	Work corev1.ObjectReference `json:"workRef"`
+
+	// +kubebuilder:default=false
+	Applied bool `json:"applied"`
+
+	// ManifestWorkStatus ManifestWorkStatus.Condition can be
+	// +kubebuilder:validation:Enum=Progressing;Available;Degraded;Applied
+	// +optional
+	ManifestWorkStatus ManifestWorkStatus `json:"status"`
```

Comment:
I was hoping to store Work* (WorkProgressing, WorkApplied, WorkAvailable or WorkDegraded from each MW. If that value is not available though, then I might removed.  Since Applied is one of the options here, removing Line 77 would make sense.

Reply:
But this approach might be to ambitious.

Reply:
I would treat these condition as orthogonal. e.g. A resource can be applied=true available=false (because someone on spoke accidentally remove it) or applied=false available=true (because the latest update on the resource fails).

We currently have not implemented progressing and degraded in manifestwork yet...
Progressing would be a transition state. e.g. deployment is rolling out, so actual replica is larger than the desired replica.
Degraded would be a condition that the resource is available, but not fully functioning. e.g. deployment resource exists on the spoke, but the actual replica is smaller than desired replica.

Reply:
I commented it out as a TODO for now based on your feedback.